app/models/customer_contact.rb

```diff
@@ -27,4 +37,11 @@ def normalize_emails_and_find_users
 
     true
   end
+
+  # TODO: Remove after `contact_type` is migrated to `kind` enum
+  before_save :sync_contact_type_and_kind
+
+  def sync_contact_type_and_kind
+    self[:kind] = self.class.kinds[self[:contact_type]]
```

Comment:
Why isn't this `self.kind`? Do we have to do this because the rails version? Or is this just a different way of doing it?

Reply:
In this case they're equivalent. I think `self[:kind]` was my default here because in the preceding data migration I had to get a value in the attribute hash bypassing the custom getter, and `self[...]` ensures you're getting the raw value. But it's not actually needed here.